### PR TITLE
gitlab oauth: consider hosted gitlab with relative path

### PR DESCRIPTION
**Pull request by OpenSourceZombie**
## Hey, I just made a Pull Request!

Fix backstage gitlab oauth callbacks to consider installations with [relative_url](https://docs.gitlab.com/ee/install/relative_url.html)
Originally reported in #8358, specifically [this comment](https://github.com/backstage/backstage/issues/8358#issuecomment-1014316566)
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — APPROVED** [review]
Oh that's needed indeed! 🎉 Thank you 👍

File: .changeset/slow-mirrors-eat.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-auth-backend': patch
+---
+
+oauth callbacks to consider installations with relative URL.
```
Comment (Rugvip) [inline .changeset/slow-mirrors-eat.md:5]:
```suggestion
Fix GitLab provider setup so that it supports GitLab installations with a path in the URL.
```

**Review by Rugvip — CHANGES_REQUESTED** [review]
Realized I'd like to tweak the changeset at bit

File: .changeset/slow-mirrors-eat.md (hunk 1 above)
Comment (OpenSourceZombie) [reply .changeset/slow-mirrors-eat.md]: Ah, your changes are more descriptive :+1:

**Review by Rugvip — APPROVED** [review]
Nice, thanks! 👍

**Pull request merged**